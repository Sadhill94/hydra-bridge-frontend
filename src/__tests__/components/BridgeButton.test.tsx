import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import BridgeButton, {
  BridgeButtonProps,
} from "../../common/components/BridgeButton/BridgeButton";

describe("The BridgeButton", () => {
  let mockCallback = jest.fn();
  let props: BridgeButtonProps;

  beforeEach(() => {
    props = {
      isConnected: true,
      inProgress: false,
      isNotEnoughBalance: false,
      isApproveReady: false,
      isDisabled: false,
      isEth: false,
      onWalletApprove: () => true,
      onWalletConnect: () => true,
      onMoveAssets: () => true,
    };
  });

  it("should show Connect wallet when wallet not connected", async () => {
    props.isConnected = false;
    const { asFragment } = render(
      <BridgeButton {...props} onWalletConnect={mockCallback} />
    );
    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(asFragment()).toMatchSnapshot();
    expect(button).toHaveTextContent("connect-wallet");
    expect(mockCallback).toHaveBeenCalledTimes(1);
  });

  it("should show loading spinner and be disabled when inProgress", async () => {
    props.inProgress = true;
    const { asFragment } = render(<BridgeButton {...props} />);
    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(asFragment()).toMatchSnapshot();
    expect(button).toHaveTextContent("common.loading");
    expect(button).toHaveAttribute("disabled");
  });

  it("should be disabled and warn for not enough balance", async () => {
    props.isNotEnoughBalance = true;
    const { asFragment } = render(<BridgeButton {...props} />);
    const button = screen.getByRole("button");

    expect(asFragment()).toMatchSnapshot();
    expect(button).toHaveAttribute("disabled");
    expect(button).toHaveTextContent("insufficient-balance");
  });

  it("should ask to input amount and be disabled when none is set", async () => {
    props.isAmountSet = false;
    const { asFragment } = render(<BridgeButton {...props} />);
    const button = screen.getByRole("button");

    expect(asFragment()).toMatchSnapshot();
    expect(button).toHaveAttribute("disabled");
    expect(button).toHaveTextContent("input-amount");
  });

  it("should fallback to input amount text when none of the conditions met", async () => {
    props.isAmountSet = true;
    const { asFragment } = render(<BridgeButton {...props} />);
    expect(asFragment()).toMatchSnapshot();
    const button = screen.getByRole("button");

    expect(asFragment()).toMatchSnapshot();
    expect(button).toHaveAttribute("disabled");
    expect(button).toHaveTextContent("input-amount");
  });

  it("should be able to bridge", async () => {
    props.isAbleToMove = true;
    props.isRouteIdSelected = true;
    props.isAmountSet = true;
    const { asFragment } = render(
      <BridgeButton {...props} onMoveAssets={mockCallback} />
    );
    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(asFragment()).toMatchSnapshot();
    expect(button).toHaveTextContent("bridge");
    expect(mockCallback).toHaveBeenCalledTimes(1);
  });

  it("should be able to approve", async () => {
    props.isAmountSet = true;
    props.isApproveReady = true;
    props.isApproved = false;
    const { asFragment } = render(
      <BridgeButton {...props} onWalletApprove={mockCallback} />
    );
    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(asFragment()).toMatchSnapshot();
    expect(button).toHaveTextContent("common.approve");
    expect(mockCallback).toHaveBeenCalledTimes(1);
  });
});
