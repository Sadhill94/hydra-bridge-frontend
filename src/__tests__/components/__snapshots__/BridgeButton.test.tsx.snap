// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The BridgeButton should ask to input amount and be disabled when none is set 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL sc-gsDKAQ hWjhNA jCJqpi"
    disabled=""
  >
    input-amount
  </button>
</DocumentFragment>
`;

exports[`The BridgeButton should be able to approve 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL sc-gsDKAQ dWDIrr jCJqpi"
  >
    common.approve
  </button>
</DocumentFragment>
`;

exports[`The BridgeButton should be able to bridge 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL sc-gsDKAQ dWDIrr jCJqpi"
  >
    bridge
  </button>
</DocumentFragment>
`;

exports[`The BridgeButton should be disabled and warn for not enough balance 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL sc-gsDKAQ hWjhNA jCJqpi"
    disabled=""
  >
    insufficient-balance
  </button>
</DocumentFragment>
`;

exports[`The BridgeButton should fallback to input amount text when none of the conditions met 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL sc-gsDKAQ hWjhNA jCJqpi"
    disabled=""
  >
    input-amount
  </button>
</DocumentFragment>
`;

exports[`The BridgeButton should show Connect wallet when wallet not connected 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL sc-gsDKAQ dWDIrr jCJqpi"
  >
    connect-wallet
  </button>
</DocumentFragment>
`;

exports[`The BridgeButton should show loading spinner and be disabled when inProgress 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL sc-gsDKAQ cokbcE jCJqpi"
    disabled=""
  >
    common.loading
  </button>
</DocumentFragment>
`;
